crates/services/shared-sequencer/src/service.rs

```diff
@@ -0,0 +1,250 @@
+    }
+}
+
+impl<S> Task<S>
+where
+    S: Signer,
+{
+    /// Fetch latest account metadata if it's not set
+    async fn ensure_account_metadata(&mut self) -> anyhow::Result<()> {
+        if self.account_metadata.is_some() {
+            return Ok(());
+        }
+        let ss = self
+            .shared_sequencer_client
+            .as_ref()
+            .expect("Shared sequencer client is not set");
+        self.account_metadata = Some(ss.get_account_meta(self.signer.as_ref()).await?);
+        Ok(())
+    }
+
+    /// Fetch previous order in the topic if it's not set
+    async fn ensure_prev_order(&mut self) -> anyhow::Result<()> {
+        if self.prev_order.is_some() {
+            return Ok(());
+        }
+        let ss = self
+            .shared_sequencer_client
+            .as_ref()
+            .expect("Shared sequencer client is not set");
+        self.prev_order = ss.get_topic().await?.map(|f| f.order);
+        Ok(())
+    }
+}
+
+#[async_trait]
+impl<S> RunnableTask for Task<S>
+where
+    S: Signer + 'static,
+{
+    async fn run(&mut self, watcher: &mut StateWatcher) -> TaskNextAction {
+        if !self.config.enabled {
+            let _ = watcher.while_started().await;
+            return TaskNextAction::Stop;
+        }
+
+        if let Err(err) = self.ensure_account_metadata().await {
+            // We don't want to spam the RPC endpoint with a lot of queries,
+            // so wait for one second before sending the next one.
+            tokio::time::sleep(Duration::from_secs(1)).await;
+            return TaskNextAction::ErrorContinue(err)
+        }
+        if let Err(err) = self.ensure_prev_order().await {
+            return TaskNextAction::ErrorContinue(err)
+        };
+
+        tokio::select! {
+            biased;
+            _ = watcher.while_started() => {
+                TaskNextAction::Stop
+            },
+            _ = self.interval.tick() => {
+                let blobs = {
+                    let mut lock = self.blobs.lock().await;
+                    core::mem::take(&mut *lock)
+                };
+                if blobs.is_empty() {
+                    tokio::time::sleep(Duration::from_secs(1)).await;
+                    return TaskNextAction::Continue;
+                };
+
+                let mut account = self.account_metadata.take().expect("Account metadata is not set");
+                let next_order = self.prev_order.map(|prev| prev.wrapping_add(1)).unwrap_or(0);
+                let ss =  self.shared_sequencer_client
+                    .as_ref().expect("Shared sequencer client is not set");
+                let blobs_bytes = postcard::to_allocvec(&blobs).expect("Failed to serialize SSBlob");
+
+                if let Err(err) = ss.send(self.signer.as_ref(), account, next_order, blobs_bytes).await {
+                    return TaskNextAction::ErrorContinue(err);
+                }
+
+                tracing::info!("Posted block to shared sequencer {blobs:?}");
```

Comment:
How many blobs we can potentially have in the collection? Asking because we may risk clogging the log with super long messages on `info` level. Maybe logging the `.len()` would suffice? Or something like range: "blobs.first().height, blobs.last().height"?

Reply:
My understanding is that the blobs posted here aren't necessarily adjacent, so logging only len or first and last isn't sufficient. In any case `SSBlob` is quite small.

```rust
pub struct SSBlob {
    block_height: BlockHeight,
    block_id: BlockId,
}
```